src/Yasumi/Provider/Argentina.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * This file is part of the Yasumi package.
+ *
+ * Copyright (c) 2015 - 2021 AzuyaLabs
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author Nader Safadi <nader@gizra.com>
+ */
+
+namespace Yasumi\Provider;
+
+use DateInterval;
+use DateTime;
+use Yasumi\Exception\InvalidDateException;
+use Yasumi\Exception\UnknownLocaleException;
+use Yasumi\Holiday;
+
+/**
+ * Provider for all holidays in Argentina.
+ */
+class Argentina extends AbstractProvider
+{
+    use CommonHolidays;
+    use ChristianHolidays;
+
+    /**
+     * Code to identify this Holiday Provider. Typically, this is the ISO3166
+     * code corresponding to the respective country or sub-region.
+     */
+    public const ID = 'AR';
+
+    /**
+     * Initialize holidays for Argentina.
+     *
+     * @throws InvalidDateException
+     * @throws \InvalidArgumentException
+     * @throws UnknownLocaleException
+     * @throws \Exception
+     */
+    public function initialize(): void
```

Comment:
This method is pretty fat. I recommend to extract the custom holidays logic into separate (private) methods to make the code more cleaner.

Reply:
I would make separate methods for each holiday, making it even more cleaner.

Reply:
Done.